insights/parsers/nvidia.py

```diff
@@ -85,3 +103,63 @@ def gpu_models(self):
         str: Returns the GPUs model set
         """
         return set([gpu["model"] for gpu in self])
+
+
+@parser(Specs.nvidia_smi_active_clocks_event_reasons)
+class NvidiaSmiActiveClocksEventReasons(Parser, list):
+    """
+    Parser for the output of command `/usr/bin/nvidia-smi --query-gpu=name,clocks_event_reasons.active --format=csv,noheader`.
+    This command lists each of the NVIDIA GPUs in the system, along with their names and bitmasks
+    of active clock event reasons.
+
+    Raises:
+        ParseException: When run into an unparsable line
+        SkipComponent: When content is empty
+
+    Sample Content::
+        NVIDIA L4, 0x0000000000000001
+        NVIDIA A1, 0x0000000000000000
+        NVIDIA H1, 0x0000000000000084
+
+    Examples::
+        >>> type(active_clocks_event_reasons)
+        <class 'insights.parsers.nvidia.NvidiaSmiActiveClocksEventReasons'>
+        >>> len(active_clocks_event_reasons)
+        3
+        >>> active_clocks_event_reasons[0]['applications_clocks_setting']
+        False
+        >>> active_clocks_event_reasons[2]['hw_power_brake_slowdown']
+        True
+        >>> active_clocks_event_reasons[2]['none']
+        False
+    """
+
+    def parse_content(self, content):
+        if not content:
+            raise SkipComponent("Empty content.")
+
+        for line in content:
+            items = line.split(",")
+            if len(items) != 2 or not items[1].strip().startswith("0x"):
+                raise ParseException(
+                    "Not an expected command output for active clocks event reasons: %s" % line
+                )
+            bitmask = int(items[1].strip().strip("LL"), 16)
+            self.append(
+                {
+                    "gpu_name": items[0].strip(),
+                    "applications_clocks_setting": APPLICATIONS_CLOCKS_SETTING & bitmask
+                    == APPLICATIONS_CLOCKS_SETTING,
+                    "display_clock_setting": DISPLAY_CLOCK_SETTING & bitmask
+                    == DISPLAY_CLOCK_SETTING,
+                    "gpu_idle": GPU_IDLE & bitmask == GPU_IDLE,
+                    "none": NONE | bitmask == NONE,
+                    "sw_power_cap": SW_POWER_CAP & bitmask == SW_POWER_CAP,
+                    "sw_thermal_slowdown": SW_THERMAL_SLOWDOWN & bitmask == SW_THERMAL_SLOWDOWN,
+                    "sync_boosst": SYNC_BOOST & bitmask == SYNC_BOOST,
+                    "hw_power_brake_slowdown": HW_POWER_BRAKE_SLOWDOWN & bitmask
+                    == HW_POWER_BRAKE_SLOWDOWN,
+                    "hw_slowdown": HW_SLOWDOWN & bitmask == HW_SLOWDOWN,
+                    "hw_thermal_slowdown": HW_THERMAL_SLOWDOWN & bitmask == HW_THERMAL_SLOWDOWN,
+                }
+            )
```

Comment:
Better make the following change which is equaling to yours to make the code a bit sample.

```python
BITMASKS = {
    "none": 0x0000000000000000,
    "gpu_idle": 0x0000000000000001,
    "applications_clocks_setting": 0x0000000000000002,
    "sw_power_cap": 0x0000000000000004,
    "hw_slowdown": 0x0000000000000008,
    "sync_boost": 0x0000000000000010,
    "sw_thermal_slowdown": 0x0000000000000020,
    "hw_thermal_slowdown": 0x0000000000000040,
    "hw_power_brake_slowdown": 0x0000000000000080,
    "display_clock_setting": 0x0000000000000100,
}

...
            data = dict(gpu_name=items[0].strip())
            for key, bm in BITMASKS.items():
                data[key] = bitmask & bm == bm
            self.append(data)
```

Reply:
Thanks, updated. Please help take a look again.